radicle-surf/src/file_system/directory.rs

```diff
@@ -165,50 +159,45 @@ pub enum DirectoryContents {
 impl DirectoryContents {
     /// Get a label for the `DirectoryContents`, either the name of the [`File`]
     /// or the name of the [`Directory`].
-    pub fn label(&self) -> Label {
+    pub fn label(&self) -> &Label {
         match self {
-            DirectoryContents::File { name, .. } => name.clone(),
+            DirectoryContents::File { name, .. } => name,
             DirectoryContents::Directory(directory) => directory.current(),
         }
     }
 }
 
-impl From<SubTree<Label, File>> for DirectoryContents {
-    fn from(sub_tree: SubTree<Label, File>) -> Self {
-        match sub_tree {
-            SubTree::Node { key, value } => DirectoryContents::File {
-                name: key,
-                file: value,
-            },
-            SubTree::Branch { key, forest } => DirectoryContents::Directory(Directory {
-                current: Location::SubDirectory(key),
-                sub_directories: (*forest).into(),
-            }),
-        }
-    }
-}
-
 impl Directory {
     /// Create a root directory.
     ///
     /// This function is usually used for testing and demonstation purposes.
     pub fn root() -> Self {
         Directory {
-            current: Location::Root,
-            sub_directories: Forest::root(),
+            name: Label::root(),
+            contents: BTreeMap::new(),
         }
     }
 
     /// Create a directory, similar to `root`, except with a given name.
     ///
     /// This function is usually used for testing and demonstation purposes.
-    pub fn new(label: Label) -> Self {
+    pub fn new(name: Label) -> Self {
         Directory {
-            current: Location::SubDirectory(label),
-            sub_directories: Forest::root(),
+            name,
+            contents: BTreeMap::new(),
         }
     }
 
+    /// get the name
```

Comment:
nit: let's follow English grammar rules and start with a capital :)
```suggestion
    /// Get the name of the current `Directory`.
```